File: CSharp Advanced/AdvancedCSharpExamProblemPract/13SumOfAllValues/SumOfAllValues.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.RegularExpressions;
using System.IO;


class SumOfAllValues
{
    static void Main(string[] args)
    {
        Console.SetIn(new StreamReader(Console.OpenStandardInput(8192)));

        string keyString = Console.ReadLine();
        string text = Console.ReadLine();

        string startKeyPattern = @"^[^\W0-9]+(?=\d)";
        string endKeyPattern = @"(?<=\d)[^\W0-9]+$";

        string startKey = Regex.Match(keyString, startKeyPattern).Value;
        string endKey = Regex.Match(keyString, endKeyPattern).Value;

         if(startKey == String.Empty || endKey == String.Empty )
        {
            Console.WriteLine("<p>A key is missing</p>");
            return;
        }

        string matcher = startKey + @"(\d*?\.*?\d+?)" + endKey;
        MatchCollection matches = Regex.Matches(text, matcher);

        double sum = 0;
        foreach (Match match in matches)
        {
            sum += double.Parse(match.Groups[1].Value);
        }
        if(sum == 0)
        {
            Console.WriteLine("<p>The total value is: <em>nothing</em></p>");
            
        }
        else
        {
            Console.WriteLine("<p>The total value is: <em>{0}</em></p>", sum);
        }
    }
}

File: extra/7. PHP-Basics-Exam-2014-12-21/PHP-Basics-Exam-2014-12-21/02. Sum-of-All-Values/sum-of-all-values.php
<?php
$keysString = $_GET['keys'];
$textString = $_GET['text'];

$startKeyRegex = "/^([a-zA-Z_]+)\d+/";
$endKeyRegex = "/\d+([a-zA-Z_]+)$/";
preg_match($startKeyRegex, $keysString, $startKey);
preg_match($endKeyRegex, $keysString, $endKey);

if (!$startKey || !$endKey) {
    echo "<p>A key is missing</p>";
} else {
    $startKey = $startKey[1];
    $endKey = $endKey[1];

    $numbersRegex = "/" . $startKey . "(.*?)" . $endKey . "/";
    preg_match_all($numbersRegex, $textString, $matches);

    $total = 0;
    foreach($matches[1] as $match) {
        if (is_numeric($match) && $matches != "") {
            $total += $match;
        }
    }
    echo "<p>The total value is: <em>";
    echo $total != 0 ? $total : 'nothing';
    echo "</em></p>";
}